core/common/buffer.hpp

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_BUFFER_HPP
+#define KAGOME_BUFFER_HPP
+
+#include <string>
+#include <vector>
+
+namespace kagome::common {
+
+  /**
+   * @brief Class represents arbitrary (including empty) byte buffer.
+   */
+  class Buffer {
+   public:
+    using iterator = std::vector<uint8_t>::iterator;
+    using const_iterator = std::vector<uint8_t>::const_iterator;
+
+    /**
+     * @brief allocates buffer of size={@param size}, filled with {@param byte}
+     */
+    explicit Buffer(size_t size, uint8_t byte);
+
+    /**
+     * @brief construct buffer from a byte vector
+     */
+    explicit Buffer(const std::vector<uint8_t> &v);
```

Comment:
Maybe also add a ctor, which accepts vector by move? Can be useful to avoid copying, if not needed